File: src/main/java/com/zippeykeys/praisebe/compat/EMod.java
package com.zippeykeys.praisebe.compat;

import com.zippeykeys.praisebe.iface.ILocalize;

import org.jetbrains.annotations.Contract;

import lombok.AllArgsConstructor;
import lombok.Getter;

import net.minecraftforge.fml.common.Loader;


@AllArgsConstructor
public enum EMod implements ILocalize{
    ;

    @Getter(onMethod_ = @Override)
    private final String name;

    @Override
    @Contract(pure = true)
    public String getPrefix(){
        return "mod";
    }

    public boolean exists(){
        return Loader.isModLoaded(name);
    }
}

File: src/main/kotlin/com/zippeykeys/praisebe/compat/EMod.kt
package com.zippeykeys.praisebe.compat

import com.zippeykeys.praisebe.iface.ILocalize
import net.minecraftforge.fml.common.Loader

enum class EMod(private val actualName: String) : ILocalize {
    ;

    override val prefix: String
        get() = "mod"

    override fun getName(): String {
        return actualName
    }

    fun exists(): Boolean {
        return Loader.isModLoaded(name)
    }
}